### PR TITLE
Dependent Mapping pattern #1293

### PR DESCRIPTION
**Pull request by Adrianp3623**
Pull request title

- Clearly and concisely describes what it does
- Refer to the issue that it solves, if available

Pull request description

Dependent Mapping pattern [#1293](https://github.com/iluwatar/java-design-patterns/issues/1293) may have been done.
Please review this pull request.

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

**Comment by Adrianp3623** [comment]
@iluwatar please review this pull request. This pull request resolve [Dependent Mapping pattern #1293](https://github.com/iluwatar/java-design-patterns/issues/1293).

File: dependent-mapping/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+layout: pattern
+title: Dependent Mapping
+folder: dependent-mapping
+permalink: /patterns/dependent-mapping/
+categories: Behavioral
+language: en
+tags:
+ - Database
+ - consistency
+---
```
Comment (iluwatar) [inline dependent-mapping/README.md:11]: Study https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute to see how the yaml frontmatter should be implemented

I recommend using category `Structural` and tags `Data access`. See https://github.com/iluwatar/java-design-patterns/wiki/07.-Categories-and-Tags

File: dependent-mapping/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+## Class diagram
+![alt text](./etc/dependent-mapping.png "Dependent Mapping")
+
+## Applicability
+Some objects naturally appear in the context of other objects. 
+Whenever you update the base album, 
+you can update the tracks in the album. 
+If these tracks are not referenced by any other table in the database, 
+you can simplify the mapping process by having the album mapper 
+perform track mapping and treat this mapping as a related mapping.
```
Comment (iluwatar) [inline dependent-mapping/README.md:25]: Use a bullet list

File: dependent-mapping/pom.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        <version>1.26.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>dependent-mapping</artifactId>
+    <version>1.26.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>16</maven.compiler.source>
+        <maven.compiler.target>16</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
```
Comment (iluwatar) [inline dependent-mapping/pom.xml:19]: Use the version from the parent pom.xml

File: dependent-mapping/pom.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>RELEASE</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (iluwatar) [inline dependent-mapping/pom.xml:26]: Let's use JUnit

File: dependent-mapping/pom.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.7.1</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (iluwatar) [inline dependent-mapping/pom.xml:32]: Hmm, why do you need this?

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Album.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+
+  /**
+   * get the id of album.
+   *
+   * @return the id of album.
+   */
+  public Long getId() {
+    return id;
+  }
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/Album.java:109]: Use Lombok where possible

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/AlbumMapper.java [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+  /**
+   * Load the specific album through the database by using findstatement and album id.
+   *
+   * @param id the id of album.
+   * @param rs the database result set.
+   * @return the album which has correct id and rs.
+   * @throws SQLException the exception of SQL.
+   */
+  public Album doLoad(Long id, ResultSet rs) throws SQLException {
+    String title = rs.getString(parameterIndex2);
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/AlbumMapper.java:103]: Prefer `var` keyword for local variables

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/AlbumMapper.java [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+  @Override
+  public void update(final MasterObj arg) throws SQLException {
+    try (PreparedStatement updateStatement = db.prepareStatement("UPDATE albums SET title = ? WHERE id = ?")) {
+      Album album = (Album) arg;
+      updateStatement.setLong(parameterIndex2, album.getId());
+      updateStatement.setString(parameterIndex1, album.getTitle());
+      updateStatement.execute();
+      updateDepObjs(album);
+    } finally {
+      System.out.println("--update master object work done--");
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/AlbumMapper.java:150]: Use a proper logger

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.h2.command.dml.Select;
+
+/**
+ * Some objects naturally appear in the context of other objects. Whenever you load or save the base album, you can load
+ * or save the tracks in the album. If these tracks are not referenced by any other table in the database,
+ * you can simplify the mapping process by having the album mapper perform track mapping and treat this mapping
+ * as a related mapping.
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java:14]: You could start with the pattern name e.g. In the dependent mapping pattern...

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+   * @param args not args need.
+   */
+  public static void main(String[] args) throws SQLException {
+    connection = DriverManager.getConnection(DB_URL);
+    albumMapper = new AlbumMapper(connection);
+    statement = connection.createStatement();
+    statement.execute("create table if not exists albums(id int not null ,title varchar(50) not null);");
+    statement.execute("create table if not exists tracks(seq int not null, albumid int not null, title varchar(50) not null);");
+    statement.execute("truncate table albums");
+    statement.execute("truncate table tracks");
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java:38]: Could move these to a separate method

File: dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    album.addDepObj(new Track("SecondTrack"));
+    albumMapper.update(album);
+    resultSet = statement.executeQuery("SELECT * from tracks");
+    displayTracks(resultSet);
+
+    System.out.println("Adding one track in the object:");
+    album.addDepObj(new Track("ThirdTrack"));
+    albumMapper.update(album);
+    resultSet = statement.executeQuery("SELECT * from tracks");
+    displayTracks(resultSet);
```
Comment (iluwatar) [inline dependent-mapping/src/main/java/com/iluwatar/dependentmapping/App.java:61]: Add comments to describe what is happening

File: dependent-mapping/src/test/java/com/iluwatar/dependentmapping/AlbumMapperTest.java [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+    String DB_URL = "jdbc:h2:~/test";
+    connection= DriverManager.getConnection(DB_URL);
+    albumMapper = new AlbumMapper(connection);
+    statement = connection.createStatement();
+    statement.execute("create table if not exists albums(id int not null ,title varchar(50) not null);");
+    statement.execute("create table if not exists tracks(seq int not null, albumid int not null, title varchar(50) not null);");
+  }
+
+  @Test
+  void findstatement() throws SQLException {
```
Comment (iluwatar) [inline dependent-mapping/src/test/java/com/iluwatar/dependentmapping/AlbumMapperTest.java:25]: Consider better naming for the test methods

File: dependent-mapping/src/test/java/com/iluwatar/dependentmapping/AlbumMapperTest.java [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+    connectDB();
```
Comment (iluwatar) [inline dependent-mapping/src/test/java/com/iluwatar/dependentmapping/AlbumMapperTest.java:26]: You could connect in before-test handler or even before-class

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**